### PR TITLE
chore: Bump Snaps packages

### PR DESCRIPTION
**Pull request by FrederikBolding**
## **Description**

Bumps all Snaps packages to their latest versions and deletes some unnecessary fetch polyfilling code. This fixes an issue where local Snaps would fail to install in the Flask flavor of mobile.

### CODE REVIEW

**Comment by owencraston** [comment]
This change progresses the issues I was previously experiencing with local snaps but does not resolve them. I am testing the [bitcoin snap](https://github.com/MetaMask/snap-bitcoin-wallet) and when I run it locally I get the following error. 

Simulator Screenshot - iPhone 13 mini - 2024-11-20 at 15 29 12

Logs:

```
[Error: Failed to fetch snap "local:http://localhost:8080": Failed to fetch "http://localhost:8080/snap.manifest.json". Status code: 404..]
```

whats even weirder is that if I navigate to http://localhost:8080/snap.manifest.json on the apps browser I see the manifest data. 

Simulator Screenshot - iPhone 13 mini - 2024-11-20 at 15 26 48

Steps: 
1. Clone Bitcoin repo
2. checkout the branch fix/allow-mobile. This will allows the dapp to install on mobile which does not have the same in page providerr.
3. run yarn install && yarn start
4. open the metamask mobile app in the simulator. Make sure the sim is running on the same machine as the local snap
5. after onboarding and creating a wallet, open the in app browser and paste the snap site url into the url box. http://localhost:3000/
6. click connect and see what happens.

**Review by owencraston — DISMISSED** [review]
I tested local snaps and the BDK (via this branch: https://github.com/MetaMask/snap-bitcoin-wallet/pull/354/files) and they both worked. The BDK snap is not working perfectly however it is not throwing errors so I imagine its an issue the the BDK implementation. Local installs seems to work well. Code also looks good to me.

File: app/core/Snaps/SnapBridge.ts
Comment (FrederikBolding) [inline app/core/Snaps/SnapBridge.ts:1]: The changes in this file are mostly copied directly from `BackgroundBridge`. The existing code was not working and would return garbage data in response to `metamask_getProviderState`

File: app/core/Snaps/permissions/permissions.ts
Comment (FrederikBolding) [inline app/core/Snaps/permissions/permissions.ts:1]: I synced the contents of this file with what is in the extension.

**Review by GuillaumeRx — APPROVED** [review]

**Review by tommasini — APPROVED** [review]

**Pull request merged**